pkg/csi/service/wcp/controller_helper.go

```diff
@@ -45,34 +45,45 @@ import (
 	"sigs.k8s.io/vsphere-csi-driver/pkg/syncer/k8scloudoperator"
 )
 
-// validateParam is a helper function used to validate the parameter name
-// received in the CreateVolume request for WCP CSI driver
+// validateCreateBlockReqParam is a helper function used to validate the parameter
+// name received in the CreateVolume request for block volumes on WCP CSI driver
 // Returns true if the parameter name is valid, false otherwise
-func validateParam(paramName, value string) bool {
+func validateCreateBlockReqParam(paramName, value string) bool {
 	return paramName == common.AttributeStoragePolicyID ||
 		paramName == common.AttributeFsType ||
 		paramName == common.AttributeAffineToHost ||
 		paramName == common.AttributeStoragePool ||
 		(paramName == common.AttributeHostLocal && strings.EqualFold(value, "true"))
 }
 
+// validateCreateFileReqParam is a helper function used to validate the parameter
+// name received in the CreateVolume request for file volumes on WCP CSI driver
+// Returns true if the parameter name is valid, false otherwise
+func validateCreateFileReqParam(paramName, value string) bool {
+	return paramName == common.AttributeStoragePolicyID ||
+		paramName == common.AttributeFsType
+}
+
 // ValidateCreateVolumeRequest is the helper function to validate
 // CreateVolumeRequest for WCP CSI driver.
 // Function returns error if validation fails otherwise returns nil.
 func validateWCPCreateVolumeRequest(ctx context.Context, req *csi.CreateVolumeRequest) error {
+	isBlockRequest := true
+	if common.IsFileVolumeRequest(ctx, req.GetVolumeCapabilities()) {
+		isBlockRequest = false
+	}
```

Comment:
You can just use:

`isBlockRequest := !common.IsFileVolumeRequest(ctx, req.GetVolumeCapabilities())`